From 81c6b12d0bfc2ff91f41f46bfd3ba94eadcf9ca8 Mon Sep 17 00:00:00 2001
From: Ben Cressey <bcressey@amazon.com>
Date: Sun, 6 Aug 2017 23:40:50 +0000
Subject: [PATCH 3/6] build binutils with TOOLS_DIR and SYSROOT_DIR

The binutils build only needs one stage, so we can write the binaries
directly to the new output path.

We disable the feature to poison system paths because we expect our
sysroot to be rooted at /usr when installing these tools through OS
packages.

Signed-off-by: Ben Cressey <bcressey@amazon.com>
---
 package/binutils/binutils.mk | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/package/binutils/binutils.mk b/package/binutils/binutils.mk
index 090065a3e4..e3feea73c4 100644
--- a/package/binutils/binutils.mk
+++ b/package/binutils/binutils.mk
@@ -61,9 +61,10 @@ endif
 BINUTILS_CONF_ENV += MAKEINFO=true
 BINUTILS_MAKE_OPTS += MAKEINFO=true
 BINUTILS_INSTALL_TARGET_OPTS = DESTDIR=$(TARGET_DIR) MAKEINFO=true install
+BINUTILS_INSTALL_TOOLS_OPTS = DESTDIR=$(TOOLS_DIR) MAKEINFO=true install
 HOST_BINUTILS_CONF_ENV += MAKEINFO=true
 HOST_BINUTILS_MAKE_OPTS += MAKEINFO=true
-HOST_BINUTILS_INSTALL_OPTS += MAKEINFO=true install
+HOST_BINUTILS_INSTALL_OPTS += DESTDIR=$(HOST_DIR) MAKEINFO=true install
 
 # Workaround a build issue with -Os for ARM Cortex-M cpus.
 # (Binutils 2.25.1 and 2.26.1)
@@ -84,11 +85,15 @@ HOST_BINUTILS_CONF_OPTS = \
 	--target=$(GNU_TARGET_NAME) \
 	--disable-shared \
 	--enable-static \
-	--with-sysroot=$(STAGING_DIR) \
-	--enable-poison-system-directories \
+	--with-sysroot=$(SYSROOT_DIR) \
 	$(BINUTILS_DISABLE_GDB_CONF_OPTS) \
 	$(BINUTILS_EXTRA_CONFIG_OPTIONS)
 
+HOST_BINUTILS_CONF_OPTS += \
+	--prefix="/usr" \
+	--sysconfdir="/etc" \
+	--localstatedir="/var"
+
 # binutils run configure script of subdirs at make time, so ensure
 # our TARGET_CONFIGURE_ARGS are taken into consideration for those
 define BINUTILS_BUILD_CMDS
@@ -138,5 +143,17 @@ define HOST_BINUTILS_FIXUP_HARDLINKS
 endef
 HOST_BINUTILS_POST_INSTALL_HOOKS += HOST_BINUTILS_FIXUP_HARDLINKS
 
+define HOST_BINUTILS_TOOLS_DIR_INSTALL
+	$(TARGET_MAKE_ENV) $(MAKE) $(BINUTILS_INSTALL_TOOLS_OPTS) -C $(@D)
+endef
+HOST_BINUTILS_POST_INSTALL_HOOKS += HOST_BINUTILS_TOOLS_DIR_INSTALL
+
+define HOST_BINUTILS_TOOLS_DIR_CLEANUP
+	$(RM) -rf $(TOOLS_DIR)/usr/lib/ldscripts
+	$(RM) -rf $(TOOLS_DIR)/usr/share
+	find $(TOOLS_DIR) -name '*.la' -delete
+endef
+HOST_BINUTILS_POST_INSTALL_HOOKS += HOST_BINUTILS_TOOLS_DIR_CLEANUP
+
 $(eval $(autotools-package))
 $(eval $(host-autotools-package))
-- 
2.21.0

